thread/thread.go

```diff
@@ -0,0 +1,104 @@
+package thread
+
+import (
+	"errors"
+	"net"
+	"github.com/kong/go-pdk/bridge"
+)
+
+type Thread struct {
+	bridge.PdkBridge
+
+	signalSocket net.Listener
+	signalConn net.Conn
+}
+
+func New(ch chan string) Thread {
+	return Thread{bridge.New(ch), nil, nil}
+}
+
+func (t *Thread) makeSignalSocket() (string, error) {
+	if t.signalSocket != nil {
+		return "", errors.New("already detached")
+	}
+
+	uaddr, err := net.ResolveUnixAddr("unix", "./signalSocket")
```

Comment:
This is literally creating a socket file called `./signalSocket`, right? As you said:

> the address is a unix path. should be unique enough (using PID? worker ID?)

(Also, this should live somewhere inside the "Kong runtime prefix" directory.)

If I'm understanding things right, given that a single worker (Lua VM) can have multiple coroutines servicing Kong request, it's possible that a single worker may have several `kong.thread.yield` in flight, so worker ID won't be enough.

If we don't want to have as many open sockets as in-flight requests, the implementation for a single-worker socket would have to be smarter, e.g. send a request identifier through the socket and make a dispatcher on the Go side that wakes up the correct goroutine. This would mean one long-lived socket per worker, so that would kill two birds with one stone, as it would also address the issue of socket creation overhead.